Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,366 @@
+        kcp_name = self._get_kubeadm_control_plane_name()
+        self._wait_kcp_ready(kcp_name)
+        self._retrieve_kubeconfig()
+
+        # Wait for workload system pods to be ready
+        print(self.kubeconfig_cs_cluster_filename)
+        wait_for_workload_pods_ready(kubeconfig_path=self.kubeconfig_cs_cluster_filename)
+
+    def delete_cluster(self, cluster_name=None, kubeconfig_filepath=None):
+        kubeconfig_cs_cluster_filename = f"kubeconfig-{cluster_name}.yaml"
+        try:
+            # Check if the cluster exists
+            check_cluster_command = f"kubectl get cluster {cluster_name} --kubeconfig {kubeconfig_filepath}"
+            result = subprocess.run(check_cluster_command, shell=True, check=True, capture_output=True, text=True)
+
+            # Proceed with deletion only if the cluster exists
+            if result.returncode == 0:
+                delete_command = f"kubectl delete cluster {cluster_name} --timeout=600s --kubeconfig {kubeconfig_filepath}"
+                self._run_subprocess(delete_command, "Timeout while deleting the cluster", shell=True)
+
+        except subprocess.CalledProcessError as error:
+            if "NotFound" in error.stderr:
+                logger.info(f"Cluster {cluster_name} not found. Skipping deletion.")
+            else:
+                raise RuntimeError(f"Error checking for cluster existence: {error}")
+
+        # Delete kind cluster
+        self.cluster = KindCluster(cluster_name)
+        self.cluster.delete()
+
+        # Remove kubeconfigs
+        if os.path.exists(kubeconfig_cs_cluster_filename):
+            os.remove(kubeconfig_cs_cluster_filename)
+        if os.path.exists(kubeconfig_filepath):
+            os.remove(kubeconfig_filepath)
+
+    def _apply_yaml_with_envsubst(self, yaml_file, error_msg):
+        try:
+            # Determine if the file is a local path or a URL
+            if os.path.isfile(yaml_file):
+                command = f"/tmp/envsubst < {yaml_file} | kubectl apply -f -"
+            elif yaml_file == "cso-infrastructure-components.yaml":
+                url = "https://github.com/SovereignCloudStack/cluster-stack-operator/releases/latest/download/cso-infrastructure-components.yaml"
+                command = f"curl -sSL {url} | /tmp/envsubst | kubectl apply -f -"
+            elif yaml_file == "cspo-infrastructure-components.yaml":
+                url = "https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/releases/latest/download/cspo-infrastructure-components.yaml"
+                command = f"curl -sSL {url} | /tmp/envsubst | kubectl apply -f -"
+            else:
+                raise ValueError(f"Unknown file or URL: {yaml_file}")
+
+            self._run_subprocess(command, error_msg, shell=True)
+        except subprocess.CalledProcessError as error:
+            raise RuntimeError(f"{error_msg}: {error}")
+
+    def _get_kubeadm_control_plane_name(self):
```

Comment:
this looks like a static method (no reference of self), but that might change once we stop polluting `os.environ`

Reply:
In this function, I am trying to get the name of the `kubeadmcontrolplane` resource because it is generated dynamically. In the next function, `_wait_kcp_ready`, the name of the `kcp` (kubeadmcontrolplane) resource is used to check whether it is ready. Once confirmed, you can proceed to retrieve the kubeconfig of the workload cluster.